src/algorithm/broadphase/brute_force.rs

```diff
@@ -0,0 +1,126 @@
+use super::*;
+use prelude::*;
+
+/// Broad phase collision detection brute force implementation.
+///
+/// Will simply do bounding box intersection tests for all shape combinations.
+#[derive(Debug, Default)]
+pub struct BruteForce;
+
+impl BruteForce {
+    /// Find all potentially colliding pairs of shapes
+    pub fn find_collider_pairs<A>(&self, shapes: &mut [A]) -> Vec<(A, A)>
+    where
+        A: HasBound + Clone,
+        A::Bound: Discrete<A::Bound>,
+    {
+        let mut pairs = Vec::<(A, A)>::default();
+        if shapes.len() <= 1 {
+            return pairs;
+        }
+
+        for left_index in 0..(shapes.len() - 1) {
+            for right_index in 1..shapes.len() {
+                if shapes[left_index]
```

Comment:
note: the index bound checks are done twice here 1) in the `for` loop condition, and 2) when you access an element. Please consider rewriting this using heavier iterators. For example:
```rust
for left_index in 0..(shapes.len() - 1) {
  let left = &shapes[left_index];
  for right in &shapes[left_index+1 ..] {
    ...
  }
}
```